lib/internal/repl.js

```diff
@@ -84,18 +90,26 @@ function setupHistory(repl, historyPath, ready) {
     if (err) {
       return ready(err);
     }
-
-    if (data) {
+    if (data && data[0] === '['
+        // Check the last two characters in case there is a trailing newline
+        && /\]/.test(data.substr(-2))) {
       try {
         repl.history = JSON.parse(data);
         if (!Array.isArray(repl.history)) {
           throw new Error('Expected array, got ' + typeof repl.history);
         }
         repl.history.slice(-repl.historySize);
```

Comment:
I know you didn't touch it but this line has no effect. It should be `repl.history = repl.history.slice(-repl.historySize);`